client/Contexts/Introspect/introspection.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+import {IntrospectionSchema, IntrospectionType, IntrospectionTypeRef,} from 'graphql';
+import {Entities, Entity, Expectations, OperationCallback} from "./ExpectationTypes";
+
+/**
+ * Transforms a field from an object representation to a GraphQL schema string representation
+ * @param type the type to transform
+ * @return the GraphQL schema string
+ */
+const transformFieldType = (type: IntrospectionTypeRef) => {
+    if (type.kind === 'LIST') {
+        return `[${transformFieldType(type.ofType)}]`;
+    } else if (type.kind === 'NON_NULL') {
+        return `${transformFieldType(type.ofType)}!`;
+    } else if (
+        type.kind === 'SCALAR' ||
+        type.kind === 'OBJECT' ||
+        type.kind === 'ENUM'
+    ) {
+        return type.name;
+    } else {
+        throw new Error(`Unsupported graphql kind ${type.kind} for ${type.name}`);
+    }
+};
+
+/**
+ * Converts an array into an object, using the element of the array indexed with `by` as the key.
+ * @param array The array to convert
+ * @param by The index of the array to use as the key
+ * @return the object
+ */
+const indexBy = <T>(array: readonly T[], by: string) => {
+    return array.reduce((acc, curr) => {
+        acc[curr[by]] = curr;
+        return acc;
+    }, {} as { [key: string]: T });
+};
+
+/**
+ * Indexes an introspection schema
+ * @param types The Types that exist on the schema
+ * @param schema The schema to index
+ * @param thing The item on the schema to use for indexing
+ * @param by The index of the array to use as the key
+ * @return the fields of the type identified by thing, indexed using by
+ */
+const indexSchema = (
+    types: { [key: string]: IntrospectionType },
+    schema: IntrospectionSchema,
+    thing: string,
+    by: string
+) => {
+    const field = schema[thing];
```

Comment:
This seems to be indexing a single "thing" from the schema, not the whole schema. Misleading function name?

Reply:
Yeah, I really struggled with what to call this function. Suggestions?